File: Java/01.Java-Basic/04.ConditionalStatementsExercise/src/MetricConverter.java
import java.util.Scanner;

public class MetricConverter {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double number = Double.parseDouble(scanner.nextLine());
        String input = scanner.nextLine();
        String output = scanner.nextLine();

        if (input.equals("mm")) {
            number = number * 1;
        } else if (input.equals("cm")) {
            number = number * 10;
        } else {
            number = number * 1000;
        }

        if (output.equals("mm")) {
            number = number / 1;
        } else if (output.equals("cm")) {
            number = number / 10;
        } else {
            number = number / 1000;
        }

        System.out.printf("%.3f", number);
    }
}

File: JavaScript/01.JavaScriptBasic/04Conditional-Statements-Exercise/src/MetricConverter.js
function solve(input) {
    let distance = Number(input[0]);
    let from = input[1];
    let to = input[2];

    if (from === "m") {
        distance *= 1000;
    } else if (from === "cm") {
        distance *= 10;
    } else {
        distance *= 1;
    }

    let output = distance;
    if (to === "m") {
        output /= 1000;
    } else if (to === "cm") {
        output /= 10;
    } else {
        output /= 1;
    }

    console.log(output.toFixed(3));
}

solve(["12","mm","m"]);
solve(["150","m","cm"]);
solve(["45","cm","mm"]);